### ISSUE TITLE
PARAM_VECTOR_IN seems not to be working

### ISSUE THREAD

**Issue by thejonan** [mlpack/mlpack#798]
Having a definition like this:

```
PARAM_VECTOR_IN(size_t, "ignore_dimensions", "The indices of dimensions to be ignored "
    "on split attempts.", "n");
```

and passing parameters like `-n 0,2` produces an exception in `CLI::ParseCommandLine(argc, argv);` call. The output is this:

```
[FATAL]: Caught exception from parsing command line:	the argument ('0, 2') for option '--ignore_dimensions' is invalid

libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: fatal error; see Log::Fatal output
Program ended with exit code: 9
```
which is rather strange, because this is very close to the example of `PARAM_VECTOR_IN` usage given in the comment next to its declaration.

**Comment by rcurtin** [comment]
I found that `PARAM_VECTOR` did not work when I wrote some tests for it.  So I did some digging in the `boost::program_options` documentation and figured out how to make it work.  But the usage actually turns out to be a little bit different than expected.  You'll need to do, e.g., `-n 0 2` or `-n 0 -n 2` (either works).  I updated the documentation to point that out.  Let me know if my fix does not work for you.

**Comment by thejonan** [comment]
Interesting, because I've tried the second option (`-n 0 -n 2`) before, and then it crashed with "duplicate entry" exception. I've just tried the first one - it doesn't crash, but reads only the first value. It could be some "specific" boost-on-Mac-via-MacPorts behavior.

Strange enough, the two new tests regarding this doesn't seem to fail!

**Comment by rcurtin** [comment]
Fascinating, can you tell me more about what you've done with the first option that doesn't work?

**Comment by thejonan** [comment]
I've made all this attempts in `det_main.cpp`, not it matters. Added this, as pointed out:
```
PARAM_VECTOR_IN(size_t, "ignore_dimensions", "The indices of dimensions to be ignored "
    "on split attempts.", "n");
```

Then tried to obtain the value like this:
```
std::vector<size_t> ignoreDims = CLI::GetParam<std::vector<size_t>>("ignore_dimensions");
```

Gives these arguments: `-n 2 3` and checking the `ignoreDims`'s value (in the debugger), it shows a vector with only one element - 2. (Yes, I've built MLPack with DEBUG=ON).

**Comment by thejonan** [comment]
I obviously need to take some rest. I've just tried it again and it worked as expected. And I've made several attempts before (but without updating from your repo. This time I did). It is also possible to be because I make two types of builds - one in an xcode generated project (via cmake) and another with classical Makefiles with OpenMP enabled compiler, so they could step on each other's toes, although (of course) I'm doing that in separate folders, but sharing the actual source tree.

Sorry, I think we can close this.
